runtime/near-vm-runner/src/near_vm_runner.rs

```diff
@@ -677,44 +695,54 @@ impl crate::runner::VM for NearVM {
         let vmmemory = memory.vm();
         let mut logic =
             VMLogic::new(ext, context, &self.config, fees_config, promise_results, &mut memory);
-
         let result = logic.before_loading_executable(method_name, code.code().len());
         if let Err(e) = result {
             return Ok(VMOutcome::abort(logic, e));
         }
 
-        let artifact = self.compile_and_load(code, cache)?;
-        let artifact = match artifact {
-            Ok(it) => it,
+        let cache = cache.unwrap_or(&NoContractRuntimeCache);
+        enum Outcome<E> {
+            Ok,
+            Abort(E),
+            AbortButNopOutcomeInOldProtocol(E),
+        }
+        let result = self.with_compiled_and_loaded(code, cache, |artifact| {
+            let result = logic.after_loading_executable(code.code().len());
+            if let Err(e) = result {
+                return Ok(Outcome::Abort(e));
+            }
+            let import = imports::near_vm::build(vmmemory, &mut logic, artifact.engine());
+            if let Err(e) = get_entrypoint_index(&*artifact, method_name) {
```

Comment:
Unrelated to this PR: I wondered why `get_entrypoint_index` is called even though the returned index isn't used. After looking into `get_entrypoint_index` I suppose it's called to validate the function signature. In that case the function name might be a bit off.